tests/python_tests/test_llm_pipeline_static.py

```diff
@@ -308,40 +354,36 @@ def end(self):
     tokenizer = Tokenizer(model_path)
     tokenized_input = tokenizer.encode(prompt)
 
-    pipe = LLMPipeline(model_path, "NPU", **config)
-    encoded_results = pipe.generate(tokenized_input, max_new_tokens=1000, ignore_eos=True, streamer=TestStreamer())
+    encoded_results = npu_model.generate(
+        tokenized_input, 
+        max_new_tokens=1000, 
+        ignore_eos=True, 
+        streamer=TestStreamer(),
+    )
 
     assert len(encoded_results.tokens[0]) == num_iters
 
 
 # FIXME: Known problem, output differs from stateful pipeline starting from 3rd prompt!
-@pytest.mark.skip(reason="JIRA-144780: Output differs from stateful pipeline")
-@pytest.mark.parametrize("config", pipeline_configs)
-@pytest.mark.parametrize("model_id", get_models_list())
 @pytest.mark.precommit
-def test_chat_generation(config, model_id):
+@pytest.mark.parametrize("llm_model", MODELS_LIST, indirect=True)
+@pytest.mark.parametrize("npu_config", PIPELINE_CONFIGS, indirect=True)
+def test_chat_generation(
+    ov_model: LLMPipeline,
+    npu_model: LLMPipeline,
+):
+    def generate_chat_history(pipe: LLMPipeline, questions):
+        pipe.start_chat()
+        chat_history = [ pipe.generate(question, max_new_tokens=50, do_sample=False) for question in questions ]
+        pipe.finish_chat()
+        return chat_history
+    
     questions = [
         '1+1=',
         'What is the previous answer?',
         'Why is the Sun yellow?',
         'What was my first question?'
     ]
 
-    _, _, model_path = download_and_convert_model(model_id)
-
-    answers_chat_mode_stateful = generate_with_chat_mode(model_path, "CPU", get_default_llm_properties(), questions)
-    answers_chat_mode_static = generate_with_chat_mode(model_path, "NPU", config, questions)
-    assert answers_chat_mode_stateful == answers_chat_mode_static, \
-        f"CPU output:\n{answers_chat_mode_stateful}\nNPU output:\n{answers_chat_mode_static}"
-
-    chat_history_stateful = generate_with_chat_history(model_path, "CPU", get_default_llm_properties(), questions)
-    messages_stateful = chat_history_stateful.get_messages()
-    chat_history_static = generate_with_chat_history(model_path, "NPU", config, questions)
-    messages_static = chat_history_static.get_messages()
-    assert messages_stateful == messages_static, f"CPU output:\n{messages_stateful}\nNPU output:\n{messages_static}"
-    
-    answers_chat_history_static = [msg["content"] for msg in messages_static if msg["role"] == "assistant"]
-    assert answers_chat_mode_static == answers_chat_history_static, (
-        f"NPU chat mode output:\n{answers_chat_mode_static}\n"
-        f"NPU chat history output:\n{answers_chat_history_static}"
-    )
+    chat_history_stateful = generate_chat_history(ov_model, questions)
+    chat_history_static = generate_chat_history(npu_model, questions)
```

Comment:
There are two different checks for chat mode (with `.start_chat()` and `.finish_chat()`) and for chat history (when ChatHistory instance is passed to generate).
And I can't find the asserts for both of these modes